cpmpy/solvers/pysat.py

```diff
@@ -39,15 +39,69 @@
 """
 from .solver_interface import SolverInterface, SolverStatus, ExitStatus
 from ..exceptions import NotSupportedError
-from ..expressions.core import Expression, Comparison, Operator, BoolVal
+from ..expressions.core import Comparison, Operator, BoolVal
 from ..expressions.variables import _BoolVarImpl, NegBoolView, boolvar
 from ..expressions.globalconstraints import DirectConstraint
+from ..transformations.linearize import canonical_comparison
 from ..expressions.utils import is_int, flatlist
 from ..transformations.decompose_global import decompose_in_tree
 from ..transformations.get_variables import get_variables
 from ..transformations.flatten_model import flatten_constraint
 from ..transformations.normalize import toplevel_list, simplify_boolean
 from ..transformations.reification import only_implies, only_bv_reifies
+from ..expressions.utils import eval_comparison
+
+
+# TODO [?] where should we put this transformation function?
+# TODO [?] are zero coefficients removed in linearize? Probably better to do here than there (right now this is actually `only_non_negative_terms`).
```

Comment:
I don't think we handle 0-coefficients at all, I hope we never make them?

Reply:
The user might make them, I guess, and then I hope they should be removed at some early stage in the transforms. Let me know if I should add a line of code to do this in flatten constraint or something.

Reply:
Since this won't cause bugs, we can resolve 0 coefficients another day.